src/renderers/WebGLRenderer.js

```diff
@@ -1389,7 +1393,7 @@ function WebGLRenderer( parameters ) {
 		object.onBeforeRender( _this, scene, camera, geometry, material, group );
 		currentRenderState = renderStates.get( scene, _currentArrayCamera || camera );
 
-		object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
+		if ( ! object.isSprite ) object.modelViewMatrix.multiplyMatrices( camera.matrixWorldInverse, object.matrixWorld );
```

Comment:
Can code in `sprite.onBeforeRender()` require this calculation?

Reply:
If `matrixWorld` would be manipulated in some way, then yes. However, there would be the same problem than before. The new `matrixWorld` would not be calculated for offscreen sprites although the value is required by `Sprite.raycast()`.

Um, tricky. An other approach to solve the problem is to introduce a third parameter `camera` to `Sprite.raycast()` and calculated the `modelViewMatrix` right there.

Reply:
Can `sprite.matrixWorld` be modified in `sprite.onBeforeRender()`?

Reply:
Well, there might be a problem if `object.matrixWorld` is changed in `onBeforeRender()`. But in this case, there is the same problem. `Sprite.raycast()` needs the `modelViewMatrix` for all sprites. Um, tricky...

Reply:
Sry, I've added the comment too late^^. I'm not sure if modifying the transformation of an `Object3D` is valid. Something like this would actually invalidate render lists.

Reply:
I'd leave the line as-is, especially if you are not sure.

Reply:
>I'm sure changing Object3D.matrixWorld in onBeforeRender() is no valid approach.

I just did a 30-second test and it works fine.
```javascript
sprite.onBeforeRender = function() { this.position.y += 0.01; this.updateMatrixWorld(); };
```
 I would not modify this line as you are doing.

Reply:
What do you mean with "works fine"? You now produce inconsistent data in your app since the transformation of the sprites is different than `projectObject()` worked with.